lib/_stream_readable.js

```diff
@@ -310,8 +310,7 @@ function chunkInvalid(state, chunk) {
 // 'readable' event will be triggered.
```

Comment:
Hm … what about this comment? It sounds like it needs to be adjusted, since it explains why we wanted `needReadable` here in the first place?

Reply:
I think that comment is actually wrong. It says

> if it's past the high water mark, we can push in some more.

However it has a check (simplified) as:

```js
return state.length < state.highWaterMark
```

I would prefer that we tackle that in another PR, as I would like to get this landed as it solves a potential memory leak situation. If you are ok I'll open an issue about it, as it needs to be investigated and explained better.

Reply:
@addaleax is this approach good for you? This should be ready to land.